spotbugs/src/main/java/edu/umd/cs/findbugs/detect/TestASM.java

```diff
@@ -81,8 +82,10 @@ public void visitMethodInsn(int opcode, String owner, String invokedName, String
     public FieldVisitor visitField(int access, String name, String desc, String signature, Object value) {
         if ((access & Opcodes.ACC_STATIC) != 0 && (access & Opcodes.ACC_FINAL) != 0 && (access & Opcodes.ACC_PUBLIC) != 0
                 && !name.equals(name.toUpperCase())) {
+            @DottedClassName
+            String className = xclass.getClassDescriptor().getDottedClassName();
             bugReporter.reportBug(new BugInstance(this, "NM_FIELD_NAMING_CONVENTION", Priorities.LOW_PRIORITY).addClass(this)
-                    .addField(this.name, name, desc, access));
+                    .addField(className, name, desc, access));
```

Comment:
This change here is key: instead of using `this.name` which is the slashed class name from ASM we're using the dotted `xclass.getClassDescriptor().getDottedClassName()`